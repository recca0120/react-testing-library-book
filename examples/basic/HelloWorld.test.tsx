import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import HelloWorld from './HelloWorld';

describe('HelloWorld Component', () => {
  test('renders with default name', () => {
    render(<HelloWorld />);
    
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toHaveTextContent('Hello, World!');
    
    const paragraph = screen.getByText('Welcome to React Testing Library');
    expect(paragraph).toBeInTheDocument();
  });

  test('renders with custom name', () => {
    render(<HelloWorld name="Alice" />);
    
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toHaveTextContent('Hello, Alice!');
  });

  test('component structure is correct', () => {
    const { container } = render(<HelloWorld name="Bob" />);
    
    expect(container.firstChild).toMatchSnapshot();
  });
});