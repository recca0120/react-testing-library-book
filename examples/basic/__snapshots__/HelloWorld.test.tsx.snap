// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld Component > component structure is correct 1`] = `
<div>
  <h1>
    Hello, 
    Bob
    !
  </h1>
  <p>
    Welcome to React Testing Library
  </p>
</div>
`;
